bamboo/bootstrap-integration-tests.sh

```diff
@@ -35,43 +27,24 @@ if [[ $USE_TERRAFORM_ZIPS == true ]]; then
   cd ..
   npm install && npm run prepare
   cd ..
-
 else
-  echo "***Deploying stack with built source"
-  npm run bootstrap
-fi
-
-echo "Locking stack for deployment $DEPLOYMENT"
-
-cd example
-set +e
-
-# Wait for the stack to be available
-node ./scripts/lock-stack.js true $DEPLOYMENT
-LOCK_EXISTS_STATUS=$?
-echo "Locking status $LOCK_EXISTS_STATUS"
-
-COUNTER=0
-while [[ $LOCK_EXISTS_STATUS == 100 ]]; do
-  if [[ $COUNTER -gt $TIMEOUT_PERIODS ]]; then
-    echo "Timed out waiting for stack to become available"
-    exit 1
+  echo "***Bootstrapping integration tests with source"
+  ## Deployment was done in the deploy-dev-integration-test-stack.sh script in a
+  ## prior job.    Run bootstrap only to cross link dependencies for the
+  ## int tests, then exit before deploying
+  if [[ $USE_CACHED_BOOTSTRAP == true ]]; then ## Change into cached cumulus dir
+    echo "*** Using cached bootstrap build dir"
+    cd /cumulus/
+    git fetch --all
+    git checkout "$GIT_SHA"
   fi
-  echo "Another build is using the ${DEPLOYMENT} stack."
-  sleep 30
-  ((COUNTER++))
-  node ./scripts/lock-stack.js true $DEPLOYMENT
-  LOCK_EXISTS_STATUS=$?
-done
-if [[ $LOCK_EXIST_STATUS -gt 0 ]]; then
-  exit 1
+  npm install
```

Comment:
Because the artifact is a statically built docker container that isn't rebuilt *each run*.   The caching time improvement is only realized if we build once outside the build process.   We're not creating a cache each build stage, we're doing it periodically from master. 

We then need to get the cache up to date with the state of the code at $GIT_SHA.   If the cache is set at master, say, from a week ago and a new package has been added, bootstrap needs to run to ensure it's there.   This is still quicker than a fresh bootstrap at each build stage. 

For example, testing this locally on the build container, runtimes with cache at the same bootstrap runs in around 15 seconds.    Running it with this morning's master against a cache from last week runs in around 1 minute due to some minor differences.   Running it against a clean build takes 5+ minutes, sometimes much more.